packages/client/src/devices/InputMediaDeviceManagerState.ts

```diff
@@ -1,16 +1,20 @@
-import { BehaviorSubject, distinctUntilChanged, Observable } from 'rxjs';
+import { BehaviorSubject, distinctUntilChanged } from 'rxjs';
 import { RxUtils } from '../store';
 
 export type InputDeviceStatus = 'enabled' | 'disabled' | undefined;
 
-export abstract class InputMediaDeviceManagerState {
+export abstract class InputMediaDeviceManagerState<C = MediaTrackConstraints> {
   protected statusSubject = new BehaviorSubject<InputDeviceStatus>(undefined);
   protected mediaStreamSubject = new BehaviorSubject<MediaStream | undefined>(
     undefined,
   );
   protected selectedDeviceSubject = new BehaviorSubject<string | undefined>(
     undefined,
   );
+  protected defaultConstraintsSubject = new BehaviorSubject<C | undefined>(
+    undefined,
+  );
```

Comment:
nit: BehaviorSubject -> ReplaySubject?

Reply:
Every subject is initialized with `undefined` state so that was the idea - still not sure if using RS over BS when initial state is `undefined` is a generally accepted good practice or just something I saw us do once and it stuck with me - it's fine either way, honestly. 😄

Reply:
We use `BehaviorSubject` to have an initial value. If we were to use `ReplaySubject`, the `getCurrentValue` wouldn't be able to return a value unless the `ReplaySubject` emits for the first time, we would have to add a fallback value there. 

I think this is a safer approach because that way, each state variable has to define its own default value, even if that's `undefined` in the majority of the cases.